packages/core/src/baseclient.ts

```diff
@@ -366,24 +359,36 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
       };
     }
 
-    const response = await send(finalEvent);
-    response.event = finalEvent;
+    try {
+      const response = await send(finalEvent);
+      response.event = finalEvent;
 
-    if (response.status === Status.RateLimit) {
-      // TODO: Handle rate limits and maintain a queue. For now, we require SDK
-      // implementors to override this method and handle it themselves.
+      if (response.status === Status.RateLimit) {
+        // TODO: Handle rate limits and maintain a queue. For now, we require SDK
+        // implementors to override this method and handle it themselves.
+      }
+      return response;
+    } catch (error) {
+      // We have a catch here since the transport can reject the request internally.
+      // If we do not catch this here, we will run into an endless loop.
+      logger.error(`${error}`);
+      return {
+        reason: `${error}`,
+        status: Status.Failed,
+      };
     }
-
-    return response;
   }
 
   /**
    * @inheritDoc
    */
   public async close(timeout?: number): Promise<boolean> {
-    return this.getBackend()
-      .getBuffer()
-      .drain(timeout);
+    return (await Promise.all([
+      this.getBackend()
+        .getTransport()
+        .close(timeout),
+      this.buffer.drain(timeout),
+    ])).reduce((prev, current) => prev && current);
```

Comment:
What kind of magic is this?

Reply:
We check that both buffers are empty and readuce (boolean) combine both buffers.